### PR TITLE
#2559. Add constructor augmentation tests. Part 6. Non-redirecting factory constructors

### PR DESCRIPTION
**Pull request by sgrekhov**
Yes, I know that there is https://github.com/dart-lang/language/issues/4025. But this PR is according to the current spec. Will update these tests if needed.

### CODE REVIEW

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A16_t04_lib.dart [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+//  ^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+    augmented(x: x, y: x);
+//  ^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  }
+}
+
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_constructors_A16_t04_lib.dart:64]: I think we usually don't have pairs of empty lines:
```suggestion
```

**Review by eernstg — COMMENTED** [review]
I think I wrote some comments here and forgot to press "Finish your review". Here we go.

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t07_lib.dart [hunk 2]
```diff
@@ -19,24 +19,24 @@
 augment library 'augmenting_constructors_A10_t07.dart';
 
 augment class C {
-  augment C(): augmented() {
-//             ^
+  augment C(): x = augmented() {
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t07_lib.dart]: Temporary removed

**Review by sgrekhov — COMMENTED** [review]
Updated. Temporary removed test with `augmented()` in the initializer's list. PTAL.

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t07_lib.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t07_lib.dart]: Thx!

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A16_t03_lib.dart [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_constructors_A16_t03.dart';
+import '../../Utils/expect.dart';
+import '../../Utils/static_type_helper.dart';
+
+augment class C {
+  augment factory C.foo(int x, [int y = 0]) {
+    // Note that for factory constructors every call of `augmented()` creates
+    // a new object. Let's don't create too much of them.
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_constructors_A16_t03_lib.dart:25]: Typo:
```suggestion
    // a new object. Let's not create too many of them.
```

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A16_t03_lib.dart [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_constructors_A16_t03.dart';
+import '../../Utils/expect.dart';
+import '../../Utils/static_type_helper.dart';
+
+augment class C {
+  augment factory C.foo(int x, [int y = 0]) {
+    augmented(0).expectStaticType<Exactly<C>>();
+    Expect.equals(0, augmented(0).x);
```
[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Augmentation-libraries/augmenting_constructors_A16_t03_lib.dart]: Thanks!

**Review by eernstg — COMMENTED** [review]
Thanks! Just a couple of typos left.

**Comment by sgrekhov** [comment]
Typos fixed.

File: LanguageFeatures/Augmentation-libraries/augmenting_constructors_A16_t03_lib.dart [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    // a new object. Let's not create too many of them.
+    var c1 = augmented(0);
+    c1.expectStaticType<Exactly<C>>();
+    Expect.equals(0, c1.x);
+    Expect.equals(0, c1.y);
+    var c2 = augmented(1, 1);
+    Expect.equals(1, c2.x);
+    Expect.equals(1, c2.y);
+    return augmented(x + 1, y + 1);
+  };
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_constructors_A16_t03_lib.dart:34]: Oh, one more thing after all: I think this semicolon is a syntax error...

**Comment by eernstg** [comment]
One more typo came up..

**Review by eernstg — COMMENTED** [review]
Another syntax error, I think...

**Pull request merged**